src/transformation/visitors/binary-expression/assignments.ts

```diff
@@ -3,20 +3,23 @@ import * as lua from "../../../LuaAST";
 import { cast } from "../../../utils";
 import { TransformationContext } from "../../context";
 import { isTupleReturnCall } from "../../utils/annotations";
-import { validateAssignment, validatePropertyAssignment } from "../../utils/assignment-validation";
+import { validateAssignment } from "../../utils/assignment-validation";
 import { createExportedIdentifier, getDependenciesOfSymbol, isSymbolExported } from "../../utils/export";
 import { createImmediatelyInvokedFunctionExpression, createUnpackCall, wrapInTable } from "../../utils/lua-ast";
 import { LuaLibFeature, transformLuaLibFunction } from "../../utils/lualib";
 import { isArrayType, isDestructuringAssignment } from "../../utils/typescript";
 import { transformElementAccessArgument } from "../access";
+import { transformLuaTablePropertyAccessInAssignment } from "../lua-table";
 import { isArrayLength, transformDestructuringAssignment } from "./destructuring-assignments";
 
 export function transformAssignmentLeftHandSideExpression(
     context: TransformationContext,
     node: ts.Expression
 ): lua.AssignmentLeftHandSideExpression {
     const symbol = context.checker.getSymbolAtLocation(node);
-    const left = context.transformExpression(node);
+    const left =
+        (ts.isPropertyAccessExpression(node) && transformLuaTablePropertyAccessInAssignment(context, node)) ||
```

Comment:
It's not the same as ternary, because when `transformLuaTablePropertyAccessInAssignment` returns `undefined` it should use `context.transformExpression`. So it's the same as
```ts
let left: lua.Expression;

if (ts.isPropertyAccessExpression(node)) {
	left = transformLuaTablePropertyAccessInAssignment(context, node);
}

if (!left) {
	left = context.transformExpression(node);
}
```

Reply:
I see, this is pretty awkward to write. I think the current code is very unclear, but your alternative is also not great. Thoughts on:
```ts
const left = ts.isPropertyAccessExpression(node)
    ? transformLuaTablePropertyAccessInAssignment(context, node) ?? context.transformExpression(node)
    : context.transformExpression(node);
```